pages/transactions/[id].js

```diff
@@ -0,0 +1,330 @@
+import Modal from "@/components/Modal";
+import TransactionForm from "@/components/TransactionForm";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+import Image from "next/image";
+import { capitalizeFirstLetter, formatDate, formatNumber } from "@/lib/utils";
+
+export default function TransactionDetailsPage({
+  transactionsList,
+  handleEditTransaction,
+  isModalOpen,
+  isEditing,
+  closeModal,
+  handleOpenEditMode,
+  handleDeleteTransaction,
+  successMessage,
+}) {
+  const router = useRouter();
+  const { id } = router.query;
+  const [transactionDetails, setTransactionDetails] = useState(null);
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  const formattedAmount = transactionDetails
+    ? formatNumber(transactionDetails)
+    : "";
+  const capitalizedType = transactionDetails
+    ? capitalizeFirstLetter(transactionDetails.type)
+    : "";
+  const formatedDate = transactionDetails
+    ? formatDate(transactionDetails.date)
+    : "";
+
+  useEffect(() => {
+    if (id && transactionsList) {
+      const transaction = transactionsList.find(
+        (transaction) => transaction.id === id
+      );
+      setTransactionDetails(transaction);
+    }
+  }, [id, transactionsList]);
+
+  function handleConfirmDelete() {
+    handleDeleteTransaction(transactionDetails.id);
+    setIsDeleting(false);
+    router.push("/");
+  }
+
+  function handleCancelDeleteDialogue() {
+    setIsDeleting(false);
+  }
+
+  function handleOpenDeleteDialogue() {
+    setIsDeleting(true);
+  }
+
+  if (!transactionDetails) {
+    return (
+      <>
+        <StyledPageNotFoundMessage>
+          Page not found
+          <StyledBackButton onClick={() => router.push("/")}>
+            <StyledImage
+              src="/images/arrow-return-left.svg"
+              alt="edit button"
+              width={15}
+              height={15}
+            />
+            Back to Transactions
+          </StyledBackButton>
+        </StyledPageNotFoundMessage>
+      </>
+    );
+  }
+
+  return (
+    <main>
+      <StyledBackButton onClick={() => router.push("/")}>
+        <StyledImage
+          src="/images/arrow-return-left.svg"
+          alt="edit button"
+          width={15}
+          height={15}
+        />
+        Back to Transactions
+      </StyledBackButton>
+
+      <Modal isModalOpen={isModalOpen} onCloseModal={closeModal}>
+        <TransactionForm
+          isEditing={isEditing}
+          initialData={transactionDetails}
+          onSubmit={(updatedTransaction) => {
+            handleEditTransaction(updatedTransaction);
+            closeModal();
+          }}
+          variant="edit"
+        />
+      </Modal>
+
+      {successMessage && (
+        <StyledSuccessMessage>{successMessage}</StyledSuccessMessage>
+      )}
```

Comment:
you should think about moving the success message into a "layout" component which is rendered on all pages. This way you don't have to include this code in every page (and might forget it on one).

Reply:
Thank you, we would prefer to include to layout on a later stage.